src/Compilers/Extension/SetGlobalProjectCollectionForCPS.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Build;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Roslyn.Compilers.Extension
+{
+    [ExportBuildGlobalPropertiesProvider]
+    [AppliesTo("(" + ProjectCapabilities.CSharp + " | " + ProjectCapabilities.VB + ")" + " & " + ProjectCapabilities.LanguageService)]
+    public class SetGlobalGlobalPropertiesForCPS : StaticGlobalPropertiesProviderBase
+    {
+        [ImportingConstructor]
+        public SetGlobalGlobalPropertiesForCPS(IProjectService projectService, ILocalRegistry2 registry, IProjectThreadingService threadingService)
```

Comment:
Is in ILocalRegistry2 in the MEF container?

Reply:
Its present at solution load, I'll need to check when this is composed

Reply:
It doesn't matter when its composed; if this get instantiated correctly then it will always be available assuming this isn't an extension that exports it, the service is exported as `SLocalRegistry` as the interface type, so I'm surprised this works.

Reply:
I'll just compute it on package load.